File: iteration2/src/human/Human.java
package iteration2.src.human;

import iteration2.src.input_output.Logger;

public abstract class Human {

    protected String firstName;
    protected String middleName;
    protected String lastName;

    protected Human(String firstName, String middleName, String lastName){
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
    }

    protected Human(String firstName, String lastName){
        this(firstName,null,lastName);
    }


    public String getFullName(){
        return firstName + ((middleName == null || middleName.length() <= 0) ? "" : " " + middleName) + " " + lastName;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }
}

File: iteration3/Human.py
class Human:
    def __init__(self, first_name, middle_name=None, last_name=None):
        self.first_name = first_name
        self.middle_name = middle_name
        self.last_name = last_name

    def get_full_name(self):
        return self.first_name + ('' if self.middle_name is None or len(self.middle_name) <= 0 else ' ' + self.middle_name) + ' ' + self.last_name

    def get_first_name(self):
        return self.first_name

    def get_middle_name(self):
        return self.middle_name

    def get_last_name(self):
        return self.last_name
